### ISSUE TITLE
[YSQL] Crash when using a function with cursors on temporary tables.

### ISSUE THREAD

**Issue by m-iancu** [yugabyte/yugabyte-db#1829]
To replicate: 

```
create temp table test as
  select n as i, n as j from generate_series(1,10) n;
create or replace function foo() returns void as $$
declare
  c cursor for select * from test;
begin
  for r in c loop
    raise notice '%, %', r.i, r.j;
  end loop;
end;
$$ language plpgsql;
select foo();
```
_Note: This works fine for non-temporary tables._

Stack trace is: 

```
#0  0x00000000004849ae in heap_copytuple (tuple=0x2fa9da8)
    at ../../../../../../../src/postgres/src/backend/access/common/heaptuple.c:741
#1  0x000000000065c251 in spi_printtup (slot=0x2fa9990, self=<optimized out>)
    at ../../../../../../src/postgres/src/backend/executor/spi.c:1860
#2  0x00000000006255ac in ExecutePlan (estate=0x2fa9568, planstate=0x2fa9780,
    use_parallel_mode=<optimized out>, operation=CMD_SELECT, sendTuples=true, numberTuples=1,
    direction=ForwardScanDirection, dest=0xd7a9d8 <spi_printtupDR>, execute_once=<optimized out>)
    at ../../../../../../src/postgres/src/backend/executor/execMain.c:1768
#3  0x000000000062541d in standard_ExecutorRun (queryDesc=0x2b78f60, direction=<optimized out>,
    count=1, execute_once=<optimized out>)
    at ../../../../../../src/postgres/src/backend/executor/execMain.c:365
#4  0x00007f40b210f3c3 in ?? ()
#5  0x0000000000dcf0f8 in error_context_stack ()
#6  0x36f3c6cef2df985e in ?? ()
#7  0x0000000000000000 in ?? ()
(gdb) quit
```

Likely root cause is that the `t_ybctid` field is not set correctly when trying to copy a tuple. For a temporary tables it needs to be set to `0` (like we do in for instance in `heap_form_tuple`) but is probably not initialized correctly in this case.

**Comment by d-uspenskiy** [comment]
Potentially duplicate of #1681

**Comment by ndeodhar** [comment]
Fixed as part of #2005
